trunner/device.py

```diff
@@ -149,9 +155,10 @@ def output(self):
         return output
 
     def kill(self):
-        os.killpg(os.getpgid(self.proc.pid), signal.SIGTERM)
+        if self.proc.isalive():
+            os.killpg(os.getpgid(self.proc.pid), signal.SIGTERM)
         self.reader_thread.join(timeout=10)
-        if self.reader_thread.is_alive():
+        if self.reader_thread.is_alive() and self.proc.isalive():
```

Comment:
No need to check if `reader_thread` is alive. It could finish without proc ending.
```python
if self.proc.isalive():
```